src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/IHoverInfoService.cs

```diff
@@ -9,4 +11,6 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Hover;
 internal interface IHoverInfoService
 {
     VSInternalHover? GetHoverInfo(string documentFilePath, RazorCodeDocument codeDocument, SourceLocation location, VSInternalClientCapabilities clientCapabilities);
+    Task<VSInternalHover?> GetHoverInfoAsync(VersionedDocumentContext versionedDocumentContext, DocumentPositionInfo positionInfo, Position position, IRazorDocumentMappingService documentMappingService, VSInternalClientCapabilities? _clientCapabilities, CancellationToken cancellationToken);
```

Comment:
As with semantic tokens, I think it would be nicer if the impl got the client capabilities from DI. I don't honestly know if that is possible with cohosting right now, so feel free to log an issue for follow up, but it would again help with making this service feel a little more well designed.

Reply:
What if we have the service implement `ICapabilitiesProvider` or have a different interface like `ICapabilitiesReader`?

Reply:
> What if we have the service implement ICapabilitiesProvider

Didn't you have a PR yesterday that specifically stopped services implementing that interface? :P

There should be an interface that can be used to get the client capabilities, I think its `IInitializeManager`, I'm just not sure if it works in cohosting as well, so was thinking we might need to wrap it with something that can unify the two worlds. `ICapabilitiesReader` would be a fine name for that latter service.

Reply:
> Didn't you have a PR yesterday that specifically stopped services implementing that interface? :P

Look... okay... 😅